AppDelegate.mm

```diff
@@ -97,6 +119,22 @@ - (void)applicationDidFinishLaunching:(NSNotification*)notification {
   NSURL* downloadURL = [NSURL URLWithString:[downloadURLs valueForKey:ARCH_KEY_NAME]];
   NSString* targetAppName = [info valueForKey:@"TargetAppName"];
 
+  if (!checkPermission()) {
+    // If permission check fails, it's likely to be a standard user, let's rerun the program
+    // as root
+    
+    if (geteuid() != 0) {
+      STPrivilegedTask* task = [STPrivilegedTask new];
+      [task setLaunchPath:[NSBundle.mainBundle executablePath]];
```

Comment:
Lets add populate `arguments` with something like `--privileged`? If we see that parameter on launch, we shouldn't try to launch the app again if the permission check fails. I don't think we necessarily need this because macOS should take care of preventing multiple instances, but lets add just in case to avoid an infinite loop of relaunching the process.

Reply:
Just confirming, you are suggesting using explicit argument as a replacement to the current `geteuid` check?